aten/src/ATen/native/vulkan/ops/Upsample.cpp

```diff
@@ -0,0 +1,80 @@
+#include <ATen/native/UpSample.h>
+#include <ATen/native/vulkan/ops/Common.h>
+#include <torch/library.h>
+
+namespace at {
+namespace native {
+namespace vulkan {
+namespace ops {
+namespace {
+
+Tensor upsample_nearest2d(
+    const Tensor& input_arg,
+    const IntArrayRef output_sizes,
+    const c10::optional<double> scales_h,
+    const c10::optional<double> scales_w) {
+  api::Context* const context = api::context();
+
+  const Tensor input = input_arg.is_vulkan() ? input_arg : input_arg.vulkan();
+  const vTensor& v_input = convert(input);
+
+  vTensor v_output{
+    context,
+    output_sizes,
+    input.options(),
+  };
+
+  api::Command::Buffer command_buffer = context->command().pool.allocate();
+  command_buffer.begin();
+  {
+    const float scale_x = compute_scales_value<float>(scales_w, input_arg.sizes()[3], output_sizes[2]);
+    const float scale_y = compute_scales_value<float>(scales_h, input_arg.sizes()[3], output_sizes[2]);
```

Comment:
I think the scales are being computed wrong here (mismatched indices), which is causing the test to fail.

Should be something like
`scale_x = input_arg.sizes()[2]/output_sizes[0]`
`scale_y = input_arg.sizes()[2]/output_sizes[1]`